crates/core/src/network/actions/custom_send.rs

```diff
@@ -50,15 +51,25 @@ pub async fn custom_send(
 pub struct SendResponseFuture {
     context: Arc<Context>,
     id: String,
+    running_time: Instant,
 }
 
 impl Future for SendResponseFuture {
     type Output = Result<String, HolochainError>;
 
     fn poll(self: Pin<&mut Self>, cx: &mut std::task::Context) -> Poll<Self::Output> {
+        self.context
+            .future_trace
+            .write()
```

Comment:
I am wary about the performance hit if we add (another) lock acquisition to each future. Would be good to either pull this in as per config setting as well (which I guess is hard to reference from here) or only acquire the lock and write elapsed time once every second or so..